lib/OpenQA/Worker/Settings.pm

```diff
@@ -31,14 +31,12 @@ sub new ($class, $instance_number = undef, $cli_options = {}) {
         $settings_file = undef;
     }
 
-    # read global settings from config
+    # read settings from config
     my %global_settings;
-    for my $section ('global', $instance_number) {
-        if ($cfg && $cfg->SectionExists($section)) {
-            for my $set ($cfg->Parameters($section)) {
-                $global_settings{uc $set} = trim $cfg->val($section, $set);
-            }
-        }
+    if ($cfg) {
+        _read_section($cfg, 'global', \%global_settings);
+        _read_section($cfg, $instance_number, \%global_settings);
+        _read_section($cfg, "class:$_", \%global_settings) for split(',', $global_settings{WORKER_CLASS} // '');
```

Comment:
Note that [Config::IniFiles](https://metacpan.org/pod/Config::IniFiles#Groups) implements subsections / Groups with a space, so it is a builtin feature. Using a colon is not wrong per se, as the ini format doesn't specify that feature, but maybe it would be nice to use the format that is supported by the module we use.

Reply:
I guess I should have requested changes here to at least discuss it...

Reply:
I just went for the first idea that was mentioned on the ticket (probably by myself). I guess it doesn't really matter, though. All we need to achieve "grouping" here is to concatenate a string - and it really didn't come to my mind to check for a library feature to help with such a basic thing.

Reply:
We had the same discussion before though for a different feature where I mentioned the group feature.